blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/TokenInfoListBuilder.java

```diff
@@ -0,0 +1,72 @@
+package com.rackspacecloud.blueflood.io;
+
+import java.util.*;
+
+public class TokenInfoListBuilder {
+
+    /**
+     * Given a metric name foo.bar.baz.qux, for prefix foo.bar
+     * (prefix foo.bar is considered level 0, baz is level 1)
+     *
+     * tokensWithNextLevelMap will accumulate all tokens at level 1 which also have subsequent next level
+     *      tokensWithNextLevelMap  -> {baz, true}
+     *
+     * enumValuesAs1Level1Map will accumulate all enum values which are at level 1.
+     * if foo.bar is an enum metric in itself with enum values of say [one, two]
+     *      enumValuesAs1Level1Map      -> {one, false}, {two, false}
+     *
+     * tokensWithEnumsAs2LevelMap will accumulate all tokens at level 1 which have enum values.
+     * if foo.bar.baz is an enum metric in itself with enum values of say [something]
+     *      tokensWithEnumsAs2LevelMap  -> {baz, true}
+     *
+     * if there is another regular metric say at level 1, say foo.bar.test (no enum values)
+     *      tokensWithEnumsAs2LevelMap  -> {baz, true}, {test, false}
+     */
+
+    private final Map<String, Boolean> tokensWithNextLevelMap = new LinkedHashMap<String, Boolean>();
+    private final Map<String, Boolean> enumValuesAs1Level1Map = new LinkedHashMap<String, Boolean>();
+    private final Map<String, Boolean> tokensWithEnumsAs2LevelMap = new LinkedHashMap<String, Boolean>();
```

Comment:
I don't completely understand why we are again storing the tokens with the next level here in a Map. It seems like you could go straight from ```metricIndexData.getTokensWithNextLevel()``` to a list of TokenInfo objects, no?